pubnub/crypto.py

```diff
@@ -68,7 +68,7 @@ def extract_random_iv(self, message, use_random_iv):
 
     def get_initialization_vector(self, use_random_iv):
         if self.pubnub_configuration.use_random_initialization_vector or use_random_iv:
-            return secrets.token_hex(16)[0:16]
+            return secrets.token_urlsafe(16)[:16]
```

Comment:
why clip to :16 characters again? I think I don't know enough about the underlying implementation to understand why we need strings here instead of bytes, but I'm fairly sure that reducing the output of 16 random BYTES from token_urlsafe to 16 url safe characters makes this again much less random. If we don't need 16 random bytes that's fine, but it the requirement is to have 16 (secure)random bytes, then I don't think this satisfies it

Reply:
It has 16 bytes but character wise it's length is 22. And in this old implementation we deal with strings. So while it's the old, deprecated implementation I don't want to update it more than necessary